File: Todobackend/src/main/java/com/example/Todobackend/entity/Task.java
package com.example.Todobackend.entity;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @NotNull(message = "Title can not be Empty")
    private String title;
    private String text;
    private boolean completed;

    public Task() {
    }

    public Task(Long id, String title, String text, boolean completed) {
        this.id = id;
        this.title = title;
        this.text = text;
        this.completed = completed;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public boolean isCompleted() {
        return completed;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }

    @Override
    public String toString() {
        return "Task{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", text='" + text + '\'' +
                ", completed=" + completed +
                '}';
    }
}

File: todofront/lib/entity/task.dart
class Task {
  int id;
  final String? title;
  final String? text;
  bool completed;

  Task({
    required this.id,
    required this.title,
    required this.text,
    this.completed = false,
  });

  factory Task.fromMap(Map taskMap) {
    return Task(
      id: taskMap['id'],
      title: taskMap['title'],
      text: taskMap['text'],
      completed: taskMap['completed'],
    );
  }

  void toggle() {
    completed = !completed;
  }
}
